lib/jekyll/url.rb

```diff
@@ -109,14 +109,14 @@ def generate_url_from_drop(template)
         replacement = self.class.escape_path(value)
 
         match.sub(":#{winner}", replacement)
-      end.gsub(%r!//!, "/".freeze)
+      end.squeeze("/")
     end
 
     # Returns a sanitized String URL, stripping "../../" and multiples of "/",
     # as well as the beginning "/" so we can enforce and ensure it.
 
     def sanitize_url(str)
-      "/" + str.gsub(%r!/{2,}!, "/").gsub(%r!\.+/|\A/+!, "")
+      "/#{str}".gsub("..", "/").gsub("./", "").squeeze("/")
```

Comment:
Using `"/#{str}"` subtly changes things, because now `.gsub` is operating on the string with the prepended `/`, rather than _just_ the input. CI passes though, so 👍

Reply:
that was my intention.. to have the input prepended with a `/` at all times and then proceed with the sanitization..
The benchmark attached, shows that its better.. :smiley: